onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -25,6 +30,9 @@ def get_next(self) -> dict:
         """generate the input data dict for ONNXinferenceSession run"""
         raise NotImplementedError
 
+    def support_batch_inference(self):
+        return False
```

Comment:
Why do you need it? If you want to do batch inference, you can make the get_next return batch input.

Reply:
The batch input is something like: {input_1: [batch_size, 3, 384, 608],  image_shape:[1,2]}
The get_next return batch input would be like: [{input_1: [1, 3, 384, 608],  image_shape:[1,2]}, {input_1: [1, 3, 384, 608],  image_shape:[1,2]}, ....]

To compose proper batch input, we still need some works. That's why I suggest to add get_batch/load_batch into DataReader. And it seems that not all the models support batch inference, therefore we need to add a function to tell whether this model supports batch inference or not.

Reply:
I've removed support_batch_inference() and use getattr(data_reader, "get_batch", None) and callable(data_reader.get_batch) instead.